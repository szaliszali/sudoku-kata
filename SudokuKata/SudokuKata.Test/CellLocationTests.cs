namespace SudokuKata.Test;

internal class CellLocationTests
{
    [Test]
    public void BlockIndex()
    {
        Approvals.VerifyAll(new Board().AllLocations().Select(l => l.BlockIndex()), "");
    }

    [Test]
    public void BlockRow()
    {
        Approvals.VerifyAll(new Board().AllLocations().Select(l => l.BlockRow()), "");
    }

    [Test]
    public void BlockCol()
    {
        Approvals.VerifyAll(new Board().AllLocations().Select(l => l.BlockCol()), "");
    }

    [Test]
    [TestCase(2, 5)]
    [TestCase(3, 10)]
    [TestCase(4, 17)]
    public void Index(int blockSize, int expectedIndex)
    {
        var board = new Board(blockSize);
        var sut = new CellLocation(board, 1, 1);

        Assert.That(sut.Index(), Is.EqualTo(expectedIndex));
    }
}
