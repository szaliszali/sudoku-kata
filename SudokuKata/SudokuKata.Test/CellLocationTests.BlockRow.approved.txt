[0] = 0
[1] = 0
[2] = 0
[3] = 0
[4] = 0
[5] = 0
[6] = 0
[7] = 0
[8] = 0
[9] = 0
[10] = 0
[11] = 0
[12] = 0
[13] = 0
[14] = 0
[15] = 0
[16] = 0
[17] = 0
[18] = 0
[19] = 0
[20] = 0
[21] = 0
[22] = 0
[23] = 0
[24] = 0
[25] = 0
[26] = 0
[27] = 1
[28] = 1
[29] = 1
[30] = 1
[31] = 1
[32] = 1
[33] = 1
[34] = 1
[35] = 1
[36] = 1
[37] = 1
[38] = 1
[39] = 1
[40] = 1
[41] = 1
[42] = 1
[43] = 1
[44] = 1
[45] = 1
[46] = 1
[47] = 1
[48] = 1
[49] = 1
[50] = 1
[51] = 1
[52] = 1
[53] = 1
[54] = 2
[55] = 2
[56] = 2
[57] = 2
[58] = 2
[59] = 2
[60] = 2
[61] = 2
[62] = 2
[63] = 2
[64] = 2
[65] = 2
[66] = 2
[67] = 2
[68] = 2
[69] = 2
[70] = 2
[71] = 2
[72] = 2
[73] = 2
[74] = 2
[75] = 2
[76] = 2
[77] = 2
[78] = 2
[79] = 2
[80] = 2
